src/AWSS3.jl

```diff
@@ -258,7 +258,7 @@ end
 s3_get_meta(a...; b...) = s3_get_meta(global_aws_config(), a...; b...)
 
 function _s3_exists_file(aws::AbstractAWSConfig, bucket, path)
-    q = Dict("prefix" => path, "delimiter" => "", "max-keys" => 1)
+    q = Dict("prefix" => path, "delimiter" => "/", "max-keys" => 1)
```

Comment:
What does this do here?

Reply:
This is required by https://github.com/JuliaCloud/AWSS3.jl/pull/289/files#diff-0622bb94c91db091998eec0f64f90430f91c0440b8a0e19fd71cdd45c82be123R266. As the [blog post](https://repost.aws/knowledge-center/iam-s3-user-specific-folder) I based the tests off uses this I think we should update the delimiter used here (tests do fail without this)

Reply:
I'll double check that there are tests in place that require this.

Reply:
just to check my own understanding: when you specify `delimiter="/"`, this has the effect of "collapsing" all subdirectories below `prefix` to a single entry, effectively allowing the user to see _which_ subdirectories exist below `path`, but not get any information about their contents.  is that right?

Reply:
> just to check my own understanding: when you specify `delimiter="/"`, this has the effect of "collapsing" all subdirectories below `prefix` to a single entry, effectively allowing the user to see _which_ subdirectories exist below `path`, but not get any information about their contents. is that right?

Your understanding is correct. Here's the documentation on the subject: https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-prefixes.html

Here's a concrete example:
```sh
❯ aws-vault exec ReadObject --region us-east-1 -- aws s3api list-objects-v2 --bucket awss3.jl.test.20230609t183606z --prefix prefix/
- Contents:
  - ETag: '"fbfa50731929244e205bbf7dc3737bcf"'
    Key: prefix/denied/secrets/top-secret
    LastModified: '2023-06-09T18:36:38+00:00'
    Size: 21
    StorageClass: STANDARD
  - ETag: '"5d41402abc4b2a76b9719d911017c592"'
    Key: prefix/granted/file
    LastModified: '2023-06-09T18:36:38+00:00'
    Size: 5
    StorageClass: STANDARD
  EncodingType: url
  IsTruncated: false
  KeyCount: 2
  MaxKeys: 1000
  Name: awss3.jl.test.20230609t183606z
  Prefix: prefix/

❯ aws-vault exec ReadObject --region us-east-1 -- aws s3api list-objects-v2 --bucket awss3.jl.test.20230609t183606z --prefix prefix/ --delimiter /
- CommonPrefixes:
  - Prefix: prefix/denied/
  - Prefix: prefix/granted/
  Delimiter: /
  EncodingType: url
  IsTruncated: false
  KeyCount: 2
  MaxKeys: 1000
  Name: awss3.jl.test.20230609t183606z
  Prefix: prefix/
```
The reason though we need to include the delimiter in these queries is that our AWS policies allowing us to list bucket content are conditional on this `s3:delimiter: ["/"]`. If you do not pass in this delimiter than the policy is violated and you cannot list the content. As `s3_exists` was already using `endswith(path, '/')` in it's logic before this PR we've already chosen using "/" as our delimiter so this change just allows us to use policies that require the delimiter to be set.

Reply:
Since a change to the delimiter in this code may violate an existing AWS IAM policy using `s3:delimiter: [""]` I'll mark this change as breaking.